nixos/modules/services/monitoring/timescale-prometheus.nix

```diff
@@ -0,0 +1,272 @@
+      };
+
+      extraOptions = mkOption {
+        default = [];
+        type = types.listOf types.str;
+        example = literalExample ''
+          [ "-async-acks"
+          ]
+        '';
+        description = ''
+          Extra command line arguments to pass to timescale-prometheus.
+        '';
+      };
+
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = (cfg.db-host != "" && cfg.db-host != "localhost") -> (cfg.db-passwordFile != null);
+        message = "When services.timescale-prometheus.db-host is changed from default, services.timescale-prometheus.db-passwordFile must be set!";
+      }
+    ];
+
+    meta.maintainers = with lib.maintainers; [ _0x4A6F ];
+
+    systemd.services.timescale-prometheus = {
+      description = "An open-source analytical platform for Prometheus metrics";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "postgresql.service" ];
+      after = [ "postgresql.service" ];
+
+      environment = {
+        TS_PROM_ASYNC_ACKS = "${toString cfg.async-acks}";
+        TS_PROM_DB_CONNECT_RETRIES = "${toString cfg.db-connect-retries}";
+        TS_PROM_DB_HOST = "${toString cfg.db-host}";
+        TS_PROM_DB_NAME = "${toString cfg.db-name}";
+        TS_PROM_DB_PORT = "${toString cfg.db-port}";
+        TS_PROM_DB_SSL_MODE = "${toString cfg.db-ssl-mode}";
+        TS_PROM_DB_USER = "${toString cfg.db-user}";
+        TS_PROM_LABELS_CACHE_SIZE = "${toString cfg.labels-cache-size}";
+        TS_PROM_LEADER_ELECTION_PG_ADVISORY_LOCK_ID = "${toString cfg.leader-election-pg-advisory-lock-id}";
+        TS_PROM_LEADER_ELECTION_PG_ADVISORY_LOCK_PROMETHEUS_TIMEOUT = "${toString cfg.leader-election-pg-advisory-lock-prometheus-timeout}";
+        TS_PROM_LEADER_ELECTION_REST = "${toString cfg.leader-election-rest}";
+        TS_PROM_LOG_LEVEL = "${toString cfg.log-level}";
+        TS_PROM_METRICS_CACHE_SIZE = "${toString cfg.metrics-cache-size}";
+        TS_PROM_MIGRATE = "${toString cfg.migrate}";
+        TS_PROM_SCHEDULED_ELECTION_INTERVAL = "${toString cfg.scheduled-election-interval}";
+        TS_PROM_TPUT_REPORT = "${toString cfg.tput-report}";
+        TS_PROM_WEB_LISTEN_ADDRESS = "${toString cfg.web-listen-address}";
+        TS_PROM_WEB_TELEMETRY_PATH = "${toString cfg.web-telemetry-path}";
+      };
+
+      script = "
+          exec ${cfg.package}/bin/timescale-prometheus \\
+            ${optionalString (cfg.db-passwordFile != null) "-db-password $(<${cfg.db-passwordFile}) \\ "}
+            ${lib.concatStringsSep " " cfg.extraOptions}
+        ";
+
+      serviceConfig = {
+        User = "timescale-prometheus";
+        Group = "timescale-prometheus";
+        DynamicUser = "yes";
+        RuntimeDirectory = "timescale-prometheus";
+        StateDirectory = "timescale-prometheus";
+        StateDirectoryMode = "0700";
+        PrivateDevices = true;
+        # Sandboxing
+        CapabilityBoundingSet = "CAP_NET_RAW CAP_NET_ADMIN";
```

Comment:
Not overly important, and I don't know for sure, but I'm going to guess that a list is preferred here:
```suggestion
        CapabilityBoundingSet = [ "CAP_NET_RAW" "CAP_NET_ADMIN" ];
```